File: class/DatabaseManager.php
<?php 

class DatabaseManager{
    private $connect;
    private static $sharedInstance_;


    public function __construct(){
        $hostname = "localhost";
        $database = "aen";
        $username = "root";
        $password = "";
        
        try{
            $this->connect = new PDO("mysql:host=".$hostname.";dbname=".$database, $username, $password);
        }catch(PDOException $e){
            exit('problème de connexion à la base');
            print_r($e);
        }
    }      

    public function connect(){
        $hostname = "localhost";
        $database = "aen";
        $username = "root";
        $password = "";

        try{
            $this->connect = new PDO("mysql:host=".$hostname.";dbname=".$database, $username, $password);
            return $this->connect;
        }catch(PDOException $e){
            exit('problème de connexion à la base');
            print_r($e);
        }
       //return new PDO("mysql:host=".$hostname.";dbname=".$database, $username, $password);
    
    }
    public static function getSharedInstance(){
        if(!isset(DatabaseManager::$sharedInstance_)){
            DatabaseManager::$sharedInstance_ = new DatabaseManager();
        }
        return DatabaseManager::$sharedInstance_;
    }

    public function exec($sql, $params){
        $statement = $this->connect->prepare($sql);
        if($statement && $statement->execute($params)){
            return true;
        }
            return false;
    }
}

?>

File: java/src/com/esgi/database/DatabaseManager.java
package com.esgi.database;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class DatabaseManager {

    private final String URL = "jdbc:mysql://localhost:3306/aen";
    private final String USER = "root";
    private final String PASSWORD = "";

    private Connection connection;

    public Connection getConnection() {
        try {
            if (connection == null || connection.isClosed()) {
                connection = DriverManager.getConnection(URL, USER, PASSWORD);
            }
            return connection;
        } catch (SQLException e) {
            e.printStackTrace();
            return null;
        }
    }
}
